buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -328,7 +328,7 @@ static void getBytes(AbstractByteBuf buf, long addr, int index, ByteBuffer dst)
     static void setBytes(AbstractByteBuf buf, long addr, int index, ByteBuf src, int srcIndex, int length) {
         buf.checkIndex(index, length);
         checkNotNull(src, "src");
-        if (AbstractByteBuf.isInvalid(srcIndex, length, src.capacity())) {
+        if (isOutOfBounds(srcIndex, length, src.capacity())) {
             throw new IndexOutOfBoundsException("srcIndex: " + srcIndex);
```

Comment:
The reason why it was done here is because `AsciiString` now uses it. I can pull it out, but I would prefer not to (just creates extra work). As I said in https://github.com/netty/netty/pull/4413#discussion_r43532383 I will port the changes to 4.0 too.